src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts

```diff
@@ -164,6 +169,28 @@ export class ReadRespondComponent implements OnInit, AfterViewChecked {
      }
       this.responseAnswers = value;
       this.stepNext(this.profileData.respondedOn);
+    } else {
+      this.showError = true;
+    }
+  }
+
+  scrollToError() {
+      const elements = document.getElementsByClassName('error-msg');
+      if (elements.length) {
+        const top = elements[0].getBoundingClientRect().y;
+        if (top) {
+          window.scrollTo({
+            top: top,
+            behavior: 'smooth',
+          });
+        }
+      }
+  }
+
+  onChange() {
```

Comment:
@Nishanthh Do we need to have this on all the inputs? Can't we have this check when submitAnswer() is called? And instead of `enableSubmitResponse` making the Submit button  active, can we always have it active and run the submitAnswer() which will check for all the responses and if its there, let the request go through. And if fields are missing, it triggers the validation message

Reply:
@Nishanthh Just answer the question here for now. I'll create a separate improvement ticket for it

Reply:
@avikt Yes to `enableSubmitResponse` i have this on all the inputs, I will get rid of that and have the validation in submit answers.

Reply:
@Nishanthh Don't have to do this now. I'll create a separate improvement ticket for it